narayana-spring-boot-core/src/main/java/dev/snowdrop/boot/narayana/core/properties/NarayanaPropertiesInitializer.java

```diff
@@ -67,6 +67,9 @@ private void setNodeIdentifier(String nodeIdentifier) {
     }
 
     private void setXARecoveryNodes(List<String> xaRecoveryNodes) {
+        if (xaRecoveryNodes.isEmpty()) {
+            xaRecoveryNodes = List.of(getPopulator(CoreEnvironmentBean.class).getNodeIdentifier());
```

Comment:
I think if the caller (ie the user via the properties) is asking for no "bottom up" recovery (xaRecoveryNodes tells narayana which orphaned XA resources it should try to recover) then this code should be respecting that choice. If you want it to be explicit you could log an info message here, but I'd just respect the config and not try to override it.

Reply:
If you do apply a property override then it would be a change in behaviour and users would need to be aware of that. Could you point me at the snowdrop documentation about this property.

Reply:
As far as I know the only doc is the readme.md. But there might exist more for the internal Redhat fork?